packages/ra-core/src/controller/saveContext/useAddMiddleware.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+import { Middleware } from './useMutationMiddlewares';
+import { useSaveContext } from './useSaveContext';
+
+/**
+ * Internal hook that registers a middleware for the save function in the current SaveContext.
+ * @param callback The middleware function.
+ */
+export const useAddMiddleware = <
```

Comment:
"middleware" is generic. Middleware for what?

I'd call it `useRegisterMutationMiddleware`